### ISSUE TITLE
Unable to pass a list of dependencies to `implementation` when using Kotlin build script.

### ISSUE THREAD

**Issue by beatbrot** [gradle/gradle#17729]
Kotlin build script does not allow list of maps as arguments of the methods `implementation`, `compile`, etc.. This is, however, possible with a Groovy build script

### Expected Behavior
It is possible to pass a `List<Map<String,String>>` as dependency for each Configuration when using Kotlin build scripts, just like it is possible with Groovy build scripts.

### Current Behavior
The following error occurs when running `gradlew -Pkt-script=true --scan` in [this project](https://github.com/beatbrot/gradle-issue-demo) (see [Build scan](https://gradle.com/s/6al6vqdeslbsk)):

```
FAILURE: Build failed with an exception.

* Where:
Build file 'C:\Users\loych\Downloads\gradle-issue-demo\kotlin-build-script\build.gradle.kts' line: 9

* What went wrong:
Cannot convert the provided notation to an object of type Dependency: [{group=com.google.guava, name=guava, version=30.1.1-jre}, {group=org.slf4j, name=slf4j-api, version=1.7.31}].
The following types/formats are supported:
  - Instances of Dependency.
  - String or CharSequence values, for example 'org.gradle:gradle-core:1.0'.
  - Maps, for example [group: 'org.gradle', name: 'gradle-core', version: '1.0'].
  - FileCollections, for example files('some.jar', 'someOther.jar').
  - Projects, for example project(':some:project:path').
  - ClassPathNotation, for example gradleApi().

Comprehensive documentation on dependency notations is available in DSL reference for DependencyHandler type.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 2s

Publishing build scan...
https://gradle.com/s/6al6vqdeslbsk
```

When running without the project property, and therefore without calling `implementation(listOf(...))` in Kotlin, the build succeedes, even though I passed a list to `implementation` [in the groovy build script](https://github.com/beatbrot/gradle-issue-demo/blob/27b8106b58f6299a7487d85760d0ed483aa6d76c/groovy-build-script/build.gradle#L8).

### Context
I use a global Groovy Expando as a poor-mans dependency catalogue. This Groovy expando has an attribute for each dependency/bundle. Dependencies are maps, while Dependency Bundles are lists of maps.

This allows me to add a whole suite of dependencies with a command like this:

```groovy
dependencies {
    implementation deps.logging
}
```

I want to switch to Kotlin build scripts and would've expected something like this to work:

```kotlin
dependencies {
    implementation(deps.getProperty("logging"))
}
```

### Steps to Reproduce 
1. Clone https://github.com/beatbrot/gradle-issue-demo
2. Run `gradlew -Pkt-script=true`

I already figured out that the Groovy build script transforms a Collection of Maps into several method calls right here:

https://github.com/gradle/gradle/blob/b67064e9c4ff9c720023964c4adc6476066d08ec/subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/dsl/dependencies/DynamicAddDependencyMethods.java#L58-L60

This does not happen for Kotlin build scripts.

### Your Environment
A PC running Windows 10, Java 11 and Gradle 7.1.1

Build scan URL: https://gradle.com/s/6al6vqdeslbsk

**Comment by melix** [comment]
You can use [version catalog bundles](https://docs.gradle.org/current/userguide/platforms.html#sec:dependency-bundles) for this.

**Comment by beatbrot** [comment]
Yes, I already mentioned this in my initial post.

Version catalogues are, however, still in preview. In addition to that, I think the Kotlin DSL still should support everything the Groovy DSL did. This would dramatically ease the migration from Groovy to Kotlin DSL.

**Comment by mlopatkin** [comment]
Thank you for your interest in Gradle!

This issue will be closed as a duplicate of #7649. Please add your +1's to that issue.
